provider/kubernetes/kubernetes_test.go

```diff
@@ -1718,14 +1759,27 @@ func TestIngressAnnotations(t *testing.T) {
 					},
 				},
 			},
+			"rewrite/api": {
+				Backend:        "rewrite/api",
+				PassHostHeader: true,
+				Routes: map[string]types.Route{
+					"/api": {
+						Rule: "ReplacePath:/",
+					},
+					"rewrite": {
+						Rule: "Host:rewrite",
+					},
+				},
+				Priority: len("/api"),
+			},
 		},
 	}
 
 	actualJSON, _ := json.Marshal(actual)
 	expectedJSON, _ := json.Marshal(expected)
 
 	if !reflect.DeepEqual(actual, expected) {
-		t.Fatalf("expected %+v, got %+v", string(expectedJSON), string(actualJSON))
+		t.Fatalf("expected %+v \n got %+v", string(expectedJSON), string(actualJSON))
```

Comment:
We should stop comparing JSON representations as they can hide differences in very subtle ways. For instance, an empty slice and a `nil` one would not be `DeepEqual` but look the same in the JSON output.

Either compare on the `actual` and `expected` structs, or use `assert.Equal()`.